r/tests/testthat/test-type.R

```diff
@@ -280,7 +280,9 @@ test_that("infer_type() gets the right type for Expression", {
   expect_equal(y$type(), infer_type(y))
   expect_equal(infer_type(y), float64())
   expect_equal(add_xy$type(), infer_type(add_xy))
-  expect_equal(infer_type(add_xy), float64())
+  # even though 10 is a float64, arrow will clamp it to the narrowest
```

Comment:
This is only because both are scalars? If the float64 were corresponding to a field in the data, I would think that downcasting to float32 would be undesirable.

Reply:
Correct.